File: src/v5/MainScreen.java
package v5;

import java.sql.Connection;
import java.sql.SQLException;

import java.util.Scanner;

public class MainScreen {
    private final String loggedInEmployeeName;
    private final int loggedInEmployeeID;

    public MainScreen(int loggedInEmployeeID,String loggedInEmployeeName){
        this.loggedInEmployeeID = loggedInEmployeeID;
        this.loggedInEmployeeName = loggedInEmployeeName;
    }
    public void mainScreen() {

        System.out.println("Welcome to the Managing Screen");
        System.out.println("Logged-in Employee: " + loggedInEmployeeName + " (ID: " + loggedInEmployeeID + ")");

        Connection dbConnection = null;
        try {
            MySQLConnector connector = new MySQLConnector();
            dbConnection = connector.getConnection();

            CustomerScreen customerControl = new CustomerScreen(dbConnection);
            EmployeeScreen employeeControl = new EmployeeScreen(dbConnection);
            Report report = new Report(dbConnection);
            ManagingScreen managingScreen = new ManagingScreen(loggedInEmployeeID,loggedInEmployeeName);

            Scanner scanner = new Scanner(System.in);
            int choice;

            do {
                System.out.println("WELCOME TO MAIN SCREEN");
                System.out.println("1. Employee Screen");
                System.out.println("2. Customer Screen");
                System.out.println("3. Reports");
                System.out.println("4. Managing Screen");
                System.out.println("5. Quit");
                System.out.print("Enter your choice: ");
                choice = scanner.nextInt();

                switch (choice) {
                    case 1 -> employeeControl.handleOptions();
                    case 2 -> customerControl.handleOptions();
                    case 3 -> report.generateReport();
                    case 4 -> managingScreen.display();
                    case 5 -> System.out.println("Exiting MAIN SCREEN. Goodbye!");
                    default -> System.out.println("Invalid choice. Please enter a valid option.");
                }
            } while (choice != 5);
            connector.closeConnection();
        } finally {
            if (dbConnection != null) {
                try {
                    dbConnection.close();
                } catch (SQLException e) {
                    e.printStackTrace();
                }
            }
        }
    }

}

File: pythonCodeConversion/electricbill/screens/main_screen.py
from controllers.mysql_connection import MySQLConnector
from screens.customer_screen import CustomerScreen
from screens.employee_screen import EmployeeScreen
from screens.report_screen import Report


class MainScreen:
    def mainScreen(self):
        db_connection = None
        try:
            connector = MySQLConnector()
            db_connection = connector.get_connection()

            customer_control = CustomerScreen(db_connection)
            employee_control = EmployeeScreen(db_connection)
            report = Report(db_connection)

            choice = None
            while choice != 4:
                print("WELCOME TO MAIN SCREEN")
                print("1. Employee Screen")
                print("2. Customer Screen")
                print("3. Reports")
                print("4. Quit")
                choice = int(input("Enter your choice: "))

                if choice == 1:
                    employee_control.handle_options()
                elif choice == 2:
                    customer_control.handle_options()
                elif choice == 3:
                    report.generate_report()
                elif choice == 4:
                    print("Exiting MAIN SCREEN. Goodbye!")
                else:
                    print("Invalid choice. Please enter a valid option.")

            connector.close_connection()
        finally:
            if db_connection is not None:
                try:
                    db_connection.close()
                except Exception as e:
                    print(e)
